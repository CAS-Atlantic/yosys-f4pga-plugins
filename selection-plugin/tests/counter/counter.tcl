yosys -import
plugin -i selection

# Import the commands from the plugins to the tcl interpreter
yosys -import

read_verilog counter.v
read_verilog -specify -lib -D_EXPLICIT_CARRY +/xilinx/cells_sim.v
read_verilog -lib +/xilinx/cells_xtra.v
hierarchy -check -auto-top

puts "List: [selection_to_tcl_list w:*]"

# Write the selection to file
set fh [open counter.txt w]
set selection_list [selection_to_tcl_list w:*]
puts $fh $selection_list
close $fh
