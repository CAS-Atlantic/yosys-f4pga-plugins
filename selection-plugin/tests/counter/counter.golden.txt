{middle/$1\cnt[1:0]} {middle/$add$counter.v:32$5_Y} {middle/$0\cnt[1:0]} middle/clk_int middle/cnt middle/out middle/clk {top/$1\cnt[1:0]} {top/$add$counter.v:14$2_Y} {top/$0\cnt[1:0]} top/ibuf_out top/ibuf_proxy_out top/clk_int_2 top/clk_int_1 top/cnt top/out top/in top/clk2 top/clk
